src/rep.rs

```diff
@@ -25,11 +26,10 @@ struct RepSocketBackend {
 
 pub struct RepSocket {
     backend: Arc<RepSocketBackend>,
-    _accept_close_handle: Option<oneshot::Sender<bool>>,
     _fair_queue_close_handle: oneshot::Sender<bool>,
     current_request: Option<PeerIdentity>,
     fair_queue: mpsc::Receiver<(PeerIdentity, Message)>,
-    binds: Vec<Endpoint>,
+    binds: HashMap<Endpoint, oneshot::Sender<bool>>,
```

Comment:
Should the `_fair_queue_close_handle` be in here too? What does that handle do?

Reply:
not really. these are two different logical things (even so the callback is the same type). binds callback is responsible for closing couroutine that is accepting new connections. completely separate from that is another couroutine that is responsible for merging messages from all incoming connections into a single queue (according to fair queue rules)